forge/sigs.rkt

```diff
@@ -863,3 +958,156 @@ Now with functional forge, do-bind is used instead
 
 (define (solution-diff s1 s2)
   (map instance-diff (Sat-instances s1) (Sat-instances s2)))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;; Seq Library  ;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+; need to provide through expander
+
+(provide isSeqOf seqFirst seqLast indsOf idxOf lastIdxOf elems inds isEmpty hasDups)
+
+(define-syntax (isSeqOf stx)
+  (syntax-parse stx
+   [isSeqOf
+   (quasisyntax/loc stx
+     (lambda (r1 d) (isSeqOfFun #,(build-source-location stx) r1 d)))]))
```

Comment:
`isSeqOfFun` expects two arguments. Where are the tests?

You might be able to write the function as `(define ((isSeqOfFun loc) r1 d) ....)` and the macro as `(quasisyntax/loc stx (isSeqOfFun #,(build-source ....)))` . That'd be better, because it saves space. But don't do it if the source locations are wrong --- the partially-applied function might not grab a location the same way a literal lambda does.